packages/server/lib/util/settings.js

```diff
@@ -97,8 +135,37 @@ module.exports = {
     return options.testingType === 'component'
   },
 
-  configFile (options = {}) {
-    return options.configFile === false ? false : (options.configFile || 'cypress.json')
+  configFile (projectRoot, options = {}) {
+    const ls = fs.readdirSync(projectRoot)
+
+    if (options.configFile === false) {
+      return false
+    }
+
+    if (options.configFile) {
+      return options.configFile
+    }
+
+    if (ls.includes('cypress.config.ts')) {
+      return 'cypress.config.ts'
+    }
+
+    if (ls.includes('cypress.config.js')) {
+      return 'cypress.config.js'
+    }
+
+    if (ls.includes('cypress.json')) {
+      return 'cypress.json'
+    }
```

Comment:
It would be nice if there was a shared const like this that our code could use throughout

```js
const configFiles = ['cypress.config.ts', 'cypress.config.js', 'cypress.json']
```

So this could look like

```js
for (const configFile of configFiles) {
  if (ls.includes(configFile)) {
    return configFile
  }
}

```

Reply:
Done in 878035d1d5cbbded3ebbed1c88cffe4e49f2b81e

Reply:
Did you intend to change the logic so that the `.js` file is now preferred over the `.ts` file?

Reply:
Actually, I even intend to error if I find `js` and `ts`.
We ultimately want to write in the config files. 
If we have a `js` and a `ts` which one do we read? which one do we write? Is it worth it to decide? I probably need to add this to this PR.

Reply:
If we need to know the status of multiple config files, instead of the previously-mentioned `reduce` + `pathExistsAsync` code, we should do something more like this:

```js
Promise.all(CYPRESS_CONFIG_FILES.map(filename => {
  const cypressConfigPath = cwd(filename)
  return fs.pathExistsAsync(cypressConfigPath).then(found => ({
    filename,
    cypressConfigPath,
    exists
  }))
})).then(result => {
  // look at all entries in `result` to figure out how to handle cases where there
  // are both js + ts, etc

  // otherwise find the first matching one and use it
})
```